src/materials/ShaderMaterial.js

```diff
@@ -66,6 +66,8 @@ function ShaderMaterial( parameters ) {
 	this.index0AttributeName = undefined;
 	this.uniformsNeedUpdate = false;
 
+	this.glslVersion = null;
```

Comment:
I suggest to introduce two new constants `THREE.GLSL1` and `THREE.GLSL3` and use the former value as the default value of `glslVersion`.

Reply:
Will do

Reply:
I've left the default as null. This is required to apply the conditional GLSL 3.0 logic to the internal shaders. This seems to be the least invasive way to fix support for GLSL 3.0 in new code without breaking old code.

Reply:
@mrdoob should review this line. To summarize: if the default `glslVersion = null` is used then backwards compatibility is retained for users not using this new feature. If `glslVersion = GLSL1` is used then a forced migration task is required for all RawMaterialShaders that use `#version 300 es`, those programs will break.